src/modules/video/theora/OggDemuxer.cpp

```diff
@@ -118,22 +121,23 @@ OggDemuxer::StreamType OggDemuxer::findStream()
 	if (streamInited)
 	{
 		eos = false;
+		streamInited = false;
 		file->seek(0);
 		ogg_stream_clear(&stream);
 		ogg_sync_reset(&sync);
 	}
 
-	streamInited = true;
 	while (true)
 	{
 		// If this page isn't at the start of a stream, we've seen all streams
-		readPage();
+		readPage(true);
```

Comment:
Maybe it'd be better for readPage to return something instead of optionally throwing an exception? And then this code could `return STREAM_UNKNOWN` if readPage returns a bad value, or something.

I only suggest it because `TheoraVideoStream`'s constructor which calls `findStream` checks its return value and throws a slightly more detailed exception, already.